kratos/containers/pointer_vector_set.h

```diff
@@ -647,10 +648,23 @@ class PointerVectorSet final
     template <class InputIterator>
     void insert(InputIterator first, InputIterator last)
     {
-        // first sorts the input iterators and make the input unique.
-        std::sort(first, last, CompareKey());
-        auto new_last = std::unique(first, last, EqualKeyTo());
-        SortedInsert(first, new_last);
+        if constexpr(std::is_assignable_v<decltype(*std::declval<InputIterator>()), decltype(*std::declval<InputIterator>())>) {
+            // first sorts the input iterators and make the input unique if the iterators are assignable
+            std::sort(first, last, CompareKey());
```

Comment:
this means that you are changing the order of the input "in place", correct? 

I am not sure if this is what you would expect. I am not sure of what is the praxis in this context. Do you have it clear?

Reply:
This is fixed with rvalue insertion.